data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/subscriptionsapi/AnyFilter.java

```diff
@@ -18,30 +18,54 @@
 import dev.knative.eventing.kafka.broker.dispatcher.Filter;
 import io.cloudevents.CloudEvent;
 import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class AnyFilter implements Filter {
 
     private static final Logger logger = LoggerFactory.getLogger(AnyFilter.class);
 
-    private final List<Filter> filters;
+    private final List<FilterCounter> filters;
+
+    private final ArrayBlockingQueue<Integer> indexSwapQueue;
+
+    private final FilterListOptimizer filterListOptimizer;
+
+    private final ReadWriteLock readWriteLock;
```

Comment:
Generally, Vertx doesn't like locks and blocking operations, I'm pretty sure with this implementation we will block the event loop and that causes basically to block any even delivery for a particular trigger